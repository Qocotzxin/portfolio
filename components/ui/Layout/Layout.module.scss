.Layout {
  width: 100vw;
  height: 100vh;

  main {
    height: 100%;
    position: relative;
    z-index: -1;
  }

  &__transitionWrapper {
    padding-top: 4rem;
    padding-bottom: 4rem;
    height: 100%;
  }
}
