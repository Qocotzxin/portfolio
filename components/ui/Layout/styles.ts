import styles from "./Layout.module.scss";

export const Classes = {
  base: styles.Layout,
  transitionWrapper: styles.Layout__transitionWrapper,
};
