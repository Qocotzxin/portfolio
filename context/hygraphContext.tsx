import { HygraphData } from "@model/hygraph";
import { createContext, FC, PropsWithChildren, useContext } from "react";

const HygraphContext = createContext<Partial<HygraphData> | undefined>(
  undefined
);

const HygraphProvider: FC<
  PropsWithChildren & { hygraphData: Partial<HygraphData> }
> = ({ children, hygraphData }) => {
  return (
    <HygraphContext.Provider value={hygraphData}>
      {children}
    </HygraphContext.Provider>
  );
};

function useHygraphContext() {
  const context = useContext(HygraphContext);

  if (context === undefined) {
    throw new Error("useHygraphContext must be used within a HygraphProvider");
  }

  return context;
}

export { HygraphProvider, useHygraphContext };
