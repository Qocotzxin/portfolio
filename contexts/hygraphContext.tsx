import { HygraphData, HygraphModel } from "@models/hygraph";
import { createContext, FC, PropsWithChildren, useContext } from "react";

const HygraphContext = createContext<Partial<HygraphData> | undefined>(
  undefined
);

const HygraphProvider: FC<PropsWithChildren & HygraphModel> = ({
  children,
  hygraphData,
}) => {
  return (
    <HygraphContext.Provider value={hygraphData}>
      {children}
    </HygraphContext.Provider>
  );
};

function useHygraphContext() {
  const context = useContext(HygraphContext);

  if (context === undefined) {
    throw new Error("useHygraphContext must be used within a HygraphProvider");
  }

  return context;
}

export { HygraphProvider, useHygraphContext };
