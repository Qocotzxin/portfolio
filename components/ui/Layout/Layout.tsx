import { Box } from "@chakra-ui/react";
import { HygraphModel } from "@models/hygraph";
import Footer from "@ui/Footer";
import NavBar from "@ui/NavBar";
import { HygraphProvider } from "contexts/hygraphContext";
import Head from "next/head";
import { FC, PropsWithChildren } from "react";

const Layout: FC<PropsWithChildren & HygraphModel> = ({
  children,
  hygraphData,
}) => {
  return (
    <Box
      w="100vw"
      h="100vh"
      className="w-screen h-screen bg-slate-50 dark:bg-gray-900"
      data-testid="Layout"
    >
      <Head>
        <title>Cristian Etchebarne</title>
        <meta charSet="utf-8" />
        {hygraphData.metaTags?.map((mt, index) => (
          <meta key={index} name={mt.name} content={mt.content} />
        ))}
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <HygraphProvider hygraphData={hygraphData}>
        <header>
          <NavBar />
        </header>

        <main id="main">
          <Box height="100%" my="16" px="12">
            {children}
          </Box>
        </main>

        <footer>
          <Footer />
        </footer>
      </HygraphProvider>
    </Box>
  );
};

export default Layout;
