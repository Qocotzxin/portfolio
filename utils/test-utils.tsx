import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
