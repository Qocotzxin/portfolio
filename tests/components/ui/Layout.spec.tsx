import { useHygraphContext } from "@contexts/hygraphContext";
import { render, renderHook } from "@utils/test-utils";
import Layout from "@ui/Layout";

describe("Layout.", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<Layout hygraphData={{}} />);
    expect(container).toMatchSnapshot();
  });

  it("Should pass hygraph data to the provider when the value is truthy.", () => {
    const { result } = renderHook(() => useHygraphContext(), {
      wrapper: ({ children }) => (
        <Layout hygraphData={{ skipLinks: [] }}>{children}</Layout>
      ),
    });

    expect(result.current).toEqual({ skipLinks: [] });
  });
});
