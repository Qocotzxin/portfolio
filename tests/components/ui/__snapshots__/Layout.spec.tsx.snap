// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout. Should match snapshot. 1`] = `
.emotion-0 {
  width: 100vw;
  height: 100vh;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: var(--chakra-sizes-16);
  -webkit-padding-start: var(--chakra-space-8);
  padding-inline-start: var(--chakra-space-8);
  -webkit-padding-end: var(--chakra-space-8);
  padding-inline-end: var(--chakra-space-8);
  padding-top: var(--chakra-space-2);
  padding-bottom: var(--chakra-space-2);
  background: var(--chakra-colors-white);
  position: fixed;
  top: 0px;
  left: 0px;
}

.chakra-ui-dark .emotion-1:not([data-theme]),
[data-theme=dark] .emotion-1:not([data-theme]),
.emotion-1[data-theme=dark] {
  background: var(--chakra-colors-gray-800);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-2);
  margin-inline-start: var(--chakra-space-2);
}

.emotion-3 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: inherit;
  font-family: var(--chakra-fonts-mono);
  font-size: var(--chakra-fontSizes-xs);
  font-weight: var(--chakra-fontWeights-bold);
  -webkit-transition: text-shadow 0.2s linear,color 0.2s linear;
  transition: text-shadow 0.2s linear,color 0.2s linear;
}

.emotion-3:hover,
.emotion-3[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:focus-visible,
.emotion-3[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-4);
  margin-inline-start: var(--chakra-space-4);
}

.emotion-5 {
  width: 64px;
  height: 32px;
  background: var(--chakra-colors-blue-700);
  border-radius: 18px;
  position: relative;
}

.chakra-ui-dark .emotion-5:not([data-theme]),
[data-theme=dark] .emotion-5:not([data-theme]),
.emotion-5[data-theme=dark] {
  border-color: var(--chakra-colors-gray-50);
}

.emotion-6 {
  border-radius: 50%;
  background: var(--chakra-colors-blue-700);
  width: 18px;
  height: 18px;
  position: absolute;
  left: 0px;
  top: 7px;
  z-index: 3;
}

.emotion-7 {
  border-radius: 50%;
  background: var(--chakra-colors-yellow-400);
  width: 18px;
  height: 18px;
  position: absolute;
  left: 6px;
  top: 7px;
  -webkit-transform: translateX(0px);
  -moz-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  -webkit-transition: -webkit-transform 0.5s;
  transition: transform 0.5s;
  z-index: 2;
}

.emotion-8 {
  width: 1em;
  height: 1em;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
  width: 16px;
  height: 16px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: var(--chakra-colors-white);
  position: absolute;
  right: 8px;
  top: 7px;
  opacity: 1;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  z-index: 1;
}

.emotion-9 {
  width: 1em;
  height: 1em;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
  width: 16px;
  height: 16px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: var(--chakra-colors-white);
  position: absolute;
  left: 8px;
  top: 7px;
  opacity: 0;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  z-index: 4;
}

.emotion-10 {
  height: 100%;
  margin-top: var(--chakra-space-16);
  margin-bottom: var(--chakra-space-16);
  -webkit-padding-start: var(--chakra-space-12);
  padding-inline-start: var(--chakra-space-12);
  -webkit-padding-end: var(--chakra-space-12);
  padding-inline-end: var(--chakra-space-12);
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  -webkit-padding-start: var(--chakra-space-8);
  padding-inline-start: var(--chakra-space-8);
  -webkit-padding-end: var(--chakra-space-8);
  padding-inline-end: var(--chakra-space-8);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  position: fixed;
  bottom: 0px;
  left: 0px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-12>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-8);
  min-width: var(--chakra-sizes-8);
  font-size: var(--chakra-fontSizes-sm);
  -webkit-padding-start: var(--chakra-space-3);
  padding-inline-start: var(--chakra-space-3);
  -webkit-padding-end: var(--chakra-space-3);
  padding-inline-end: var(--chakra-space-3);
  color: var(--chakra-colors-blue-200);
  background: var(--chakra-colors-transparent);
  padding: 0px;
}

.emotion-15:focus-visible,
.emotion-15[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-15[disabled],
.emotion-15[aria-disabled=true],
.emotion-15[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-15:hover,
.emotion-15[data-hover] {
  background: rgba(144, 205, 244, 0.12);
}

.emotion-15:hover[disabled],
.emotion-15[data-hover][disabled],
.emotion-15:hover[aria-disabled=true],
.emotion-15[data-hover][aria-disabled=true],
.emotion-15:hover[data-disabled],
.emotion-15[data-hover][data-disabled] {
  background: initial;
}

.emotion-15:active,
.emotion-15[data-active] {
  background: rgba(144, 205, 244, 0.24);
}

.emotion-16 {
  width: 1em;
  height: 1em;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: currentColor;
}

.emotion-17 {
  z-index: 1;
}

.emotion-18 {
  outline: 2px solid transparent;
  outline-offset: 2px;
  background: var(--chakra-colors-gray-700);
  box-shadow: var(--chakra-shadows-dark-lg);
  color: inherit;
  min-width: var(--chakra-sizes-3xs);
  padding-top: var(--chakra-space-2);
  padding-bottom: var(--chakra-space-2);
  z-index: 1;
  border-radius: var(--chakra-radii-md);
  border-width: 1px;
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
}

<div>
  <div
    class="w-screen h-screen bg-slate-50 dark:bg-gray-900 emotion-0"
    data-testid="Layout"
  >
    <header>
      <nav
        class="emotion-1"
      >
        <div
          class="chakra-stack emotion-2"
        >
          <a
            class="chakra-link emotion-3"
            href="/"
          >
            <svg
              aria-labelledby="title"
              height="48"
              role="presentation"
              viewBox="0 0 48 48"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title"
                lang="en"
              >
                Logo
              </title>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="gradient"
                x1="0"
                x2="0"
                y1="0"
                y2="100%"
              >
                <stop
                  offset="40%"
                  stop-color="#4FD1C5"
                />
                <stop
                  offset="70%"
                  stop-color="#4299E1"
                />
              </lineargradient>
              <circle
                class="outterCircle"
                cx="24"
                cy="24"
                data-testid="NavLogo-outter-circle"
                fill="none"
                r="20"
                stroke="#4FD1C5"
                stroke-dasharray="128"
                stroke-dashoffset="128"
                stroke-width="1"
              />
              <circle
                class="innerCircle"
                cx="24"
                cy="24"
                data-testid="NavLogo-inner-circle"
                fill="none"
                r="16"
                stroke="#4299E1"
                stroke-dasharray="128"
                stroke-dashoffset="128"
                stroke-width="1"
              />
              <text
                class="text textAnimation"
                data-testid="NavLogo-text"
                fill="url(#gradient)"
                x="14"
                y="32"
              >
                CE
              </text>
            </svg>
          </a>
        </div>
        <div
          class="chakra-stack emotion-4"
        >
          <button
            class="emotion-5"
          >
            <div
              class="emotion-6"
            />
            <div
              class="emotion-7"
            />
            <svg
              class="chakra-icon chakra-icon stars emotion-8"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.657 6.247c.11-.33.576-.33.686 0l.645 1.937a2.89 2.89 0 0 0 1.829 1.828l1.936.645c.33.11.33.576 0 .686l-1.937.645a2.89 2.89 0 0 0-1.828 1.829l-.645 1.936a.361.361 0 0 1-.686 0l-.645-1.937a2.89 2.89 0 0 0-1.828-1.828l-1.937-.645a.361.361 0 0 1 0-.686l1.937-.645a2.89 2.89 0 0 0 1.828-1.828l.645-1.937zM3.794 1.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387A1.734 1.734 0 0 0 4.593 5.69l-.387 1.162a.217.217 0 0 1-.412 0L3.407 5.69A1.734 1.734 0 0 0 2.31 4.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387A1.734 1.734 0 0 0 3.407 2.31l.387-1.162zM10.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732L9.1 2.137a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L10.863.1z"
              />
            </svg>
            <svg
              class="chakra-icon chakra-icon cloud emotion-9"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.473 9a4.5 4.5 0 0 0-8.72-.99A3 3 0 0 0 3 14h8.5a2.5 2.5 0 1 0-.027-5z"
              />
              <path
                d="M14.544 9.772a3.506 3.506 0 0 0-2.225-1.676 5.502 5.502 0 0 0-6.337-4.002 4.002 4.002 0 0 1 7.392.91 2.5 2.5 0 0 1 1.17 4.769z"
              />
            </svg>
          </button>
        </div>
      </nav>
    </header>
    <main
      id="main"
    >
      <div
        class="emotion-10"
      />
    </main>
    <footer>
      <div
        class="emotion-11"
      >
        <div
          class="chakra-stack emotion-12"
        />
        <div
          class="chakra-stack emotion-12"
        >
          <div
            class="emotion-14"
            data-testid="LanguageSelector"
          >
            <button
              aria-controls="menu-list-:r1:"
              aria-expanded="false"
              aria-haspopup="menu"
              class="chakra-button chakra-menu__menu-button emotion-15"
              data-testid="FloatingMenu-button"
              id="menu-button-:r1:"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon emotion-16"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95a15.65 15.65 0 00-1.38-3.56A8.03 8.03 0 0118.92 8zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56A7.987 7.987 0 015.08 16zm2.95-8H5.08a7.987 7.987 0 014.33-3.56A15.65 15.65 0 008.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95a8.03 8.03 0 01-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                />
              </svg>
            </button>
            <div
              class="emotion-17"
              style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
            >
              <div
                aria-orientation="vertical"
                class="chakra-menu__menu-list emotion-18"
                id="menu-list-:r1:"
                role="menu"
                style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
